File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/ui/activity/SplashActivity.java


package com.example.javademogithubpractice.ui.activity;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;

import androidx.annotation.NonNull;

import com.example.javademogithubpractice.inject.component.AppComponent;
import com.example.javademogithubpractice.inject.component.DaggerActivityComponent;
import com.example.javademogithubpractice.inject.module.ActivityModule;
import com.example.javademogithubpractice.mvp.contract.ISplashContract;
import com.example.javademogithubpractice.mvp.presenter.SplashPresenter;
import com.example.javademogithubpractice.ui.base.BaseActivity;


public class SplashActivity extends BaseActivity<SplashPresenter> implements ISplashContract.View {

    private final String TAG = "SplashActivity";

    @Override
    public void showMainPage() {
        delayFinish();
        Uri dataUri = getIntent().getData();
        if (dataUri == null) {
            startActivity(new Intent(getActivity(), MainActivity.class));
        } else {
            //BrowserFilterActivity.handleBrowserUri(getActivity(), dataUri);
        }
    }

    @Override
    public void showLoginPage() {
        delayFinish();
        startActivity(new Intent(getActivity(), LoginActivity.class));
    }

    @Override
    protected void setupActivityComponent(AppComponent appComponent) {
        DaggerActivityComponent.builder()
                .appComponent(appComponent)
                .activityModule(new ActivityModule(getActivity()))
                .build()
                .inject(this);
    }

    @Override
    protected int getContentView() {
        return 0;
    }

    @Override
    protected void initActivity() {
        super.initActivity();
        mPresenter.getUser();
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        super.initView(savedInstanceState);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @NonNull Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
    }

}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/ui/activity/SplashActivity.kt
package com.example.kotlindemogithubproject.ui.activity

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Toast
import com.example.kotlindemogithubproject.R
import com.example.kotlindemogithubproject.inject.component.AppComponent
import com.example.kotlindemogithubproject.mvp.contract.ISplashContract
import com.example.kotlindemogithubproject.mvp.presenter.SplashPresenter
import com.example.kotlindemogithubproject.ui.base.BaseActivity


class SplashActivity : BaseActivity<SplashPresenter>(), ISplashContract.View {

    private val TAG = "SplashActivity"

    override fun getContentView(): Int {
        return 0
    }

    override fun showMainPage() {
        //delayFinish()
//        val dataUri = intent.data
//        if (dataUri == null) {
//            Toast.makeText(applicationContext, "YEEEEEESSS", Toast.LENGTH_SHORT).show()
//            //startActivity(new Intent(getActivity(), MainActivity.class));
//        } else {
//            //BrowserFilterActivity.handleBrowserUri(getActivity(), dataUri);
//        }
    }

    override fun showLoginPage() {
        //delayFinish()
        startActivity(Intent(activity, LoginActivity::class.java))
    }

    override fun setupActivityComponent(appComponent: AppComponent) {
        DaggerActivityComponent.builder()
            .appComponent(appComponent)
            .build()
            .inject(this)
    }

    override fun initActivity() {
        super.initActivity()
        mPresenter!!.getUser()
    }

    override fun initView(savedInstanceState: Bundle?) {
        super.initView(savedInstanceState)
    }

//    protected fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent) {
//        super.onActivityResult(requestCode, resultCode, data)
//
//        when (requestCode) {
//            //            case REQUEST_ACCESS_TOKEN:
//            //                if(resultCode == RESULT_OK){
//            //                    showMainPage();
//            //                }
//            //                break;
//            else -> {
//            }
//        }
//    }

}
